File: en Python/exercice2_5.py
import os
import time
import random

debut = time.time()
fils1 = os.fork()
if fils1 == 0:
    random.seed(os.getpid())
    n = random.randint(1, 10)
    time.sleep(n)
    os._exit(0)

fils2 = os.fork()
if fils2 == 0:
    random.seed(os.getpid())
    n = random.randint(1, 10)
    time.sleep(n)
    os._exit(0)
# le père attend les deux fils
os.wait()
os.wait()
print("Duree totale =", int(time.time() - debut), "secondes")

File: En C/exercice2_5.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <sys/wait.h>
#include <time.h>

int main()
{
    int n ;
    time_t debut = time(NULL) ;
    pid_t fils1 = fork() , fils2 ;
    if (fils1 == 0)
    {
        srand (getpid ());
        n = rand()%10 +1 ;
        sleep(n) ;
        exit(0) ;
    }
    fils2 = fork() ;
    if (fils2 == 0)
    {
        srand (getpid ());
        n = rand()%10 +1 ;
        sleep(n) ;
        exit(0) ;
    }
    wait(NULL);
    wait(NULL);
    printf ("Duree totale = %ld secondes\n" , time(NULL) - debut ) ;
    return(0) ;
}

